File: tree_primer/tree_primer__make_triangles.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;
using Lst = System.Collections.Generic.List<int>;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        int t = 0;
        var g = new Lst[n];

        foreach( int i in Range( 0, n ) ) {
            g[i] = new Lst( n );
        }
        foreach( int _ in Range( 0, n - 1 ) ) {
            int[] a = ReadLine().Split().Select( x => int.Parse( x ) - 1 ).ToArray();

            g[a[0]].Add( a[1] );
            g[a[1]].Add( a[0] );
        }
        foreach( Lst l in g ) {
            t += l.Count * ( l.Count - 1 ) / 2;
        }
        WriteLine( t % 2 != 0 ? "paiza" : "erik" );
        return;
    }
}

File: tree_primer/tree_primer__make_triangles.py
# 実行時間 10ms
n = int(input())
t = 0
g = [[] for _ in [0] * n]

for _ in [0] * (n - 1):
    a, b = list(map(int, input().split()))

    a -= 1
    b -= 1
    g[a].append(b)
    g[b].append(a)
for l in g:
    t += len(l) * (len(l) - 1) // 2
print("paiza" if t % 2 != 0 else "erik")
